modules/seqexec/server/src/main/scala/seqexec/server/gmos/Gmos.scala

```diff
@@ -150,19 +151,23 @@ abstract class Gmos[F[_]: Sync, T<:GmosController.SiteDependentTypes](controller
       SequenceConfiguration.calcStepType(config)
     }
 
-  override def observe(config: Config): SeqObserveF[F, ImageFileId, ObserveCommandResult] =
-    Reader { fileId =>
-      SeqActionF.liftF(calcObserveTime(config)).flatMap { x =>
-        SeqActionF.embedF(controller.observe(fileId, x))
+  override def observe(config: Config): Kleisli[F, ImageFileId, ObserveCommandResult] =
+    Kleisli { fileId =>
```

Comment:
In the future we might want to absorb the `ImageFileId` into the effect and demand `G[_]: ApplicativeAsk[?[_], ImageFileId]`, and then our return type here could be `G[ObserveCommandResult]`. At the boundary with this code you would instantiate with `Kleisli[F` and run it with the image id, yielding a plain `F` again. Just a thought for the future, this is fine for now.